static/panes/compiler.js

```diff
@@ -1457,11 +1473,13 @@ function getAsmInfo(opcode) {
 Compiler.prototype.onMouseMove = function (e) {
     if (e === null || e.target === null || e.target.position === null) return;
     if (this.settings.hoverShowSource === true && this.assembly) {
+        this.clearLinkedLines();
         var hoverAsm = this.assembly[e.target.position.lineNumber - 1];
         if (hoverAsm) {
             // We check that we actually have something to show at this point!
-            this.eventHub.emit('editorSetDecoration', this.sourceEditorId, hoverAsm.source && !hoverAsm.source.file ?
-                hoverAsm.source.line : -1, false);
+            var sourceLine = hoverAsm.source && !hoverAsm.source.file ? hoverAsm.source.line : -1;
+            this.eventHub.emit('editorLinkLine', this.sourceEditorId, sourceLine, false);
+            this.eventHub.emit('panesLinkLine', this.id, sourceLine, false, this);
```

Comment:
Passing `this` in panesLinkLine event to not highlight ourselves is the bit I was concerned about @mattgodbolt

Reply:
Oh I see! Hmm yes: that might be an issue. If we pass references to `eventHub` stuff then we rule out multiple browser windows. Inside GL it supports "pop out" from a pane into a separate window. I've always had it disabled, but communication between windows is limited to strings for JS security reasons. Thus, so far, holding references between things is something I avoided. 

This has made for the ugly messging  we have to do, sending strings/json objects around instead of just passing object references and calling methods between various panes. So...this _might_ be a problem. Will return to this; just wanted to exaplin why currently only simple objects and not references are passed around.

Reply:
There is an alternative to this (Which I was going to initially use) and it's to send the name the pane identifies as (The one we use on the GL headers), which afaik, it's unique enough for this pourpose

Reply:
Understood; that could work too. But isn't there an ID we can use? What am I missing? :) I'm being slow today/...

Reply:
The ID is unique only between panes of the same type (So there's IR id 1, as well as opt id 1 etc...)